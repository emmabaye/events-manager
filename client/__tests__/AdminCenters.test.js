import ConnectedAdminCenters, { AdminCenters } from '../components/AdminCenters.jsx';

describe('AdminCenters Component', () => {
  const initialState = {
    centerReducer: {
      status: "Error",
      message: "",
      allCenters: {
        data: [{
          id: '1',
          name: 'City Hall',
          description: {
            substr: () => {}
          },
          location: {
            substr: () => {}
          },
          capacity: "500",
          price: "50000"
        }]
      }
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const page = {
    firstPage: 1,
    currentPage: 2,
    nextPage: 3,
    lastPage: 6
  }

  const center = {
    id: 1,
    name: "Moments Event Center",
    description: "Best events centee for indoor and outdoor events",
    location: "Port Harcourt",
    capacity: 500,
    price: 200000,
    facilities: "Toilet, car park",
    userId: 1,
    image: "http://res.cloudinary.com/emmabaye/raw/upload/v1527629332/mfgudrbriui17hnqnh4y",
    available: true,
    createdAt: "2018-05-29 22:28:52.389+01",
    updatedAt: "2018-05-29 22:28:52.389+01"
  }

  const props = {
    dispatch: () => {},
    show: () => {},
    allCenters: {
      data: {
        rows: [
          center
        ],
        page
      }
    }
  };

  describe('AdminCenters form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<AdminCenters {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedAdminCenters /></Provider>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
