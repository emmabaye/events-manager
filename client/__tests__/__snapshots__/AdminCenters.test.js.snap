// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminCenters Component AdminCenters form  should render self it should render for dumb component 1`] = `
<div
  className="panel centers"
  id="centers"
>
  <div
    className="container events"
    id="events"
  >
    <div
      className="row event-row"
    >
      <Center
        centerDetails={
          Object {
            "available": true,
            "capacity": 500,
            "createdAt": "2018-05-29 22:28:52.389+01",
            "description": "Best events centee for indoor and outdoor events",
            "facilities": "Toilet, car park",
            "id": 1,
            "image": "http://res.cloudinary.com/emmabaye/raw/upload/v1527629332/mfgudrbriui17hnqnh4y",
            "location": "Port Harcourt",
            "name": "Moments Event Center",
            "price": 200000,
            "updatedAt": "2018-05-29 22:28:52.389+01",
            "userId": 1,
          }
        }
        key="1"
        show={[Function]}
      />
    </div>
    <Pagination
      currentPage={2}
      dispatch={[Function]}
      firstPage={1}
      getItems={[Function]}
      lastPage={6}
      nextPage={3}
    />
  </div>
</div>
`;
